test/server/server_test.cc

```diff
@@ -1334,6 +1334,13 @@ TEST_P(ServerInstanceImplTest, MutexContentionEnabled) {
   EXPECT_NO_THROW(initialize("test/server/test_data/server/empty_bootstrap.yaml"));
 }
 
+TEST_P(ServerInstanceImplTest, CoreDumpEnabled) {
```

Comment:
Can you add a no dump enabled test and check for the warning log? Should be very easy.

/wait

Reply:
yep yep, just pushed that

Reply:
[ you meant: dump enabled _but_ failed to actually enable it ]

Reply:
I did have to make `PlatformImpl` a unique_ptr passable via the constructor, to be able to make `enableCoreDump()` fail... hopefully it's not too bad.

Reply:
Err, I thought you mean a test case for dump enabled but prctl call fails -- so I added a test for that. Which I think is good, we want that path too.

Note that, there is no warning when core dump is not enabled... only when it's enabled but we fail to actually enabled. Did you initially meant you wanted a warn when it's not enabled? [ see the changes i just pushed ]

Reply:
Sorry yes I mistyped.